src/irc_bridge.py

```diff
@@ -0,0 +1,154 @@
+        """Return instance of pebble service.
+
+        Returns:
+            instance of pebble service or none.
+        """
+        return getattr(self._charm, "pebble_service", None)
+
+    def _on_collect_status(self, event: ops.CollectStatusEvent) -> None:
+        """Collect status event handler.
+
+        Args:
+            event: Collect status event.
+        """
+        # pylint:disable=duplicate-code
+        if not self._charm_state.synapse_config.enable_irc_bridge:
+            return
+        if self._charm_state.irc_bridge_datasource is None:
+            ops.MaintenanceStatus("Waiting for irc bridge db relation")
+            return
+        container = self._charm.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self._charm.unit.status = ops.MaintenanceStatus("Waiting for Synapse pebble")
+            return
+        irc_service = container.get_services(IRC_SERVICE_NAME)
+        if irc_service:
+            logger.debug("%s service already exists, skipping", IRC_SERVICE_NAME)
+            return
+        self.enable_irc_bridge()
+        event.add_status(ops.ActiveStatus())
+
+    def enable_irc_bridge(self) -> None:
+        """Enable irc service.
+
+        The required steps to enable the IRC bridge are:
+         - Create the IRC bridge configuration file.
+         - Create the IRC bridge registration file.
+         - Generate a PEM file for the IRC bridge.
+         - Finally, add IRC bridge pebble layer.
+
+        """
+        container = self._charm.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self._charm.unit.status = ops.MaintenanceStatus("Waiting for Synapse pebble")
+            return
+        self._charm.model.unit.status = ops.MaintenanceStatus("Configuring IRC bridge")
+        server_name = self._charm_state.synapse_config.server_name
+        db_connect_string = self._get_db_connection()
+        synapse.create_irc_bridge_config(
+            container=container, server_name=server_name, db_connect_string=db_connect_string
+        )
+        synapse.create_irc_bridge_app_registration(container=container)
+        self._create_pem_file(container=container)
+        self._pebble_service.replan_irc_bridge(container)
+        self._pebble_service.restart_synapse(container)
+        self._charm.model.unit.status = ops.ActiveStatus()
+
+    def _create_pem_file(self, container: ops.model.Container) -> None:
+        """Create a PEM file for the IRC bridge.
+
+        Args:
+            container: The container to create the PEM file in.
+
+        Raises:
+            PEMCreateError: If the PEM file creation fails.
+        """
+        self._charm.model.unit.status = ops.MaintenanceStatus("Creating PEM file for IRC bridge")
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            "[[ -f /data/config/passkey.pem ]] || "
```

Comment:
Minor: naming the file with something related to IRC bridge? 
And BTW What happens if this file is lost?

Reply:
Renamed, thanks!
If the file is lost, a new one can be generated, this is used for storing the irc password for the server, the pass is supplied as a matrix command in the admin room. Still this could be covered by our backup and restore, though I feel this is a minor thing to address now.